### PR TITLE
Don't automerge PRs with package downgrades

### PR DESCRIPTION
**Pull request by JohnTortugo**
Add merge policy that prevents auto merging PRs that contains package version downgrades.

@mmitche I implemented the idea that we discussed offline: add the list of required updates to the InProgressPullRequest object. To make that work I had to do a bunch of 'refactoring':

- Add the DependencyUpdateSummary class; This way we only serialize the information that we need.
- Had to create IPullRequest interface so that IMergePolicyEvaluatorContext could contain a 'PullRequest' property. 
- Small change in MergePolicies.csproj so that it can reference Maestro.Contracts.csproj
- Some changes to convert "prUrl" to "pr.Url"
- Some changes to work with DependencyUpdate instead of DependencyDetail

Tested locally + a private repo in Azdo: https://dnceng.visualstudio.com/internal/_git/Cesar2/pullrequest/6499

### CODE REVIEW

**Comment by JohnTortugo** [comment]
This now is ready for review. PTAL.

**Comment by JohnTortugo** [comment]
@mmitche - can you PTAL?

/cc @riarenas @epananth

File: src/Maestro/Maestro.MergePolicies/DontAutomergeDowngradesMergePolicy.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            {
+                context.Succeed("No version downgrade detected.");
+            }
+        }
+
+        private static bool HasAnyDowngrade(IPullRequest pr)
+        {
+            foreach (var dependency in pr.RequiredUpdates)
+            {
+                SemanticVersion.TryParse(dependency.FromVersion, out var fromVersion);
```
Comment (mmitche) [inline src/Maestro/Maestro.MergePolicies/DontAutomergeDowngradesMergePolicy.cs:37]: What happens if this fails to parse?

Comment (JohnTortugo) [reply src/Maestro/Maestro.MergePolicies/DontAutomergeDowngradesMergePolicy.cs]: Added a check for that.

**Pull request merged**